### PR TITLE
Generic Conditional Concurrency feature Proposal cum Design

### PR DESCRIPTION
**Pull request by SrinivasChilveri**
What this PR does / why we need it:
This PR is a new proposal cum design for the Generic Condition Dynamic Concurrency

### CODE REVIEW

File: content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    commands:
+      - echo run-job-b
+  Job-c:
+    cool: nginx:latest
+    genericcondition: 
+       dependjobname: Job-a
+       matchrules:
+        - key: testscenarioinv
+          operator: Exists
+            - testwithInOp
```
Comment (hzxuzhonghu) [inline content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md]: values ?

Comment (hzxuzhonghu) [reply content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md]: or should this just for key, and no value ?

File: content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      - echo run-job-c
+  Job-d:
+    tool: nginx:latest
+    genericcondition: 
+       dependjobname: Job-a
+       matchrules:
+         - key: testscenario
+           operator: DoesNotExists
+           values:
+             - testwithInOp
```
Comment (hzxuzhonghu) [inline content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md]: ditto

File: content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md (hunk 1 above)
Comment (SrinivasChilveri) [reply content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md]: yes, no values for Exists /DoesNotExists  operators

File: content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md (hunk 2 above)
Comment (SrinivasChilveri) [reply content/docs/design/generic-conditional-concurrency/generic-conditional-concurrency.md]: fixed

**Pull request merged**